src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -718,16 +718,28 @@ def retrieveBlocks(self):
                 updateBlocksDAO.execute(blocks = loadedBlocks,
                                         conn = myThread.transaction.conn,
                                         transaction = True)
-            except WMException:
-                myThread.transaction.rollback()
-                raise
             except Exception as ex:
                 myThread.transaction.rollback()
-                msg =  "Unhandled exception while finished closed blocks in DBSBuffer\n"
-                msg += str(ex)
-                logging.error(msg)
-                logging.debug("Blocks for Update: %s\n" % loadedBlocks)
-                raise DBSUploadException(msg)
+                # possible deadlock with PhEDExInjector, retry once after 10s
```

Comment:
how about adding a logging.warning here just to log a simple message (and saying it will be retried in 10secs)?